### PR TITLE
Fix width and height issues that would cut off content in `gr.DataFrame`

### PR DESCRIPTION
**Pull request by abidlabs**
This is an attempt to fix: https://github.com/gradio-app/gradio/issues/5592. Previously, if you would run a demo like `matrix_transpose` or code like this:

```py
import gradio as gr

df = [[f"col: {i} -- row:{j}" for j in range(1, 3)] for i in range(1, 5)]

with gr.Blocks() as demo:
    gr.DataFrame(df, headers=list(range(1, 11)))
    gr.DataFrame(df, headers=list(range(1, 11)), interactive=True)

if __name__ == "__main__":
    demo.launch()
```

you'd end up with a compressed dataframe whose content was cut off and scrollbars would always be present:

image

This made small dataframes frustrating to use. I think the issue was that the scrollbar width/height was not taken into account. Now, both are taken into account. The scrollbar holders are always present, but the scroll bars themselves only appear when needed:

image

Would appreciate a review @pngwn to ensure that this doesn't break anything else and this is the correct approach.

### CODE REVIEW

File: js/dataframe/shared/Table.svelte [hunk 1]
```diff
@@ -529,7 +529,7 @@
 
 	function set_cell_widths(): void {
 		const widths = cells.map((el, i) => {
-			return el?.clientWidth || 0;
+			return el?.offsetHeight || 0;
```
Comment (pngwn) [inline js/dataframe/shared/Table.svelte:532]: Why are we measuring the height to get the width of the cell?

File: js/dataframe/shared/VirtualTable.svelte [hunk 2]
```diff
@@ -56,14 +56,18 @@
 		end = i;
 		const remaining = _items.length - end;
 
+		const scrollbarHeight = viewport.offsetHeight - viewport.clientHeight;
+		if (scrollbarHeight > 0) {
+			content_height += scrollbarHeight;
+		}
+
```
Comment (pngwn) [inline js/dataframe/shared/VirtualTable.svelte:63]: The only issue with this is that a scrollbars are often dynamic, hiding and showing in response to interaction.

File: js/dataframe/shared/Table.svelte (hunk 1 above)
Comment (abidlabs) [reply js/dataframe/shared/Table.svelte:532]: Whoops my bad, will fix

File: js/dataframe/shared/VirtualTable.svelte (hunk 2 above)
Comment (abidlabs) [reply js/dataframe/shared/VirtualTable.svelte:63]: From my tests, this didn’t appear to be an issue. Does this not work for you?

Comment (pngwn) [reply js/dataframe/shared/VirtualTable.svelte:63]: I've only reviewed code so far, will test properly in the morning.

Comment (abidlabs) [reply js/dataframe/shared/VirtualTable.svelte:63]: sg thanks

**Comment by abidlabs** [comment]
I also updated the python docstrings. To avoid confusion, I set the default `height` to `500` directly in the Python class constructor.

**Review by pngwn — APPROVED** [review]
Tested this now and the code looks good, nice solution! Thanks for taking care of this @abidlabs!

**Comment by abidlabs** [comment]
Thanks @pngwn for looking it over, much appreciated.

**Pull request merged**